src/test/java/edu/unc/lib/boxc/migration/cdm/services/ArchivalDestinationsServiceTest.java

```diff
@@ -138,4 +139,129 @@ public void collectionNumberToPidTest() throws Exception {
                 Arrays.stream(solrValues.get(0).getFilterQueries()).findFirst().get());
     }
+
+    @Test
+    public void newDestinationsFileWithPid() throws Exception {
+        solrResponseWithPid();
+        Path destinationMappingsPath = project.getDestinationMappingsPath();
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+
+        service.addArchivalCollectionMappings(options);
+        assertTrue(Files.exists(destinationMappingsPath));
+
+        try (
+                Reader reader = Files.newBufferedReader(destinationMappingsPath);
+                CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                        .withFirstRecordAsHeader()
+                        .withHeader(FieldUrlAssessmentService.URL_CSV_HEADERS)
+                        .withTrim());
+        ) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertEquals("", rows.get(0).get(1));
+            assertEquals("40147", rows.get(1).get(1));
+        }
+    }
+
+    @Test
+    public void newDestinationsFileWithoutPid() throws Exception {
+        solrResponseWithoutPid();
+        Path destinationMappingsPath = project.getDestinationMappingsPath();
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+
+        service.addArchivalCollectionMappings(options);
+
+        try (
+                Reader reader = Files.newBufferedReader(destinationMappingsPath);
+                CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                        .withFirstRecordAsHeader()
+                        .withHeader(FieldUrlAssessmentService.URL_CSV_HEADERS)
+                        .withTrim());
+        ) {
+            List<CSVRecord> rows = csvParser.getRecords();
+            assertEquals("", rows.get(0).get(1));
+            assertEquals("", rows.get(1).get(1));
+        }
+    }
+
+    @Test
+    public void existingDestinationsFileWithoutForceFlag() throws Exception {
+        solrResponseWithPid();
+        writeCsv(mappingBody("aid:40126,bdbd99af-36a5-4bab-9785-e3a802d3737e,"));
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+
+        try {
+            service.addArchivalCollectionMappings(options);
+            fail();
+        } catch (StateAlreadyExistsException e) {
+            assertTrue(e.getMessage().contains("Cannot create destinations, a file already exists."
+                    + " Use the force flag to overwrite."));
+        }
+    }
+
+    @Test
+    public void existingDestinationsFileWithForceFlag() throws Exception {
+        solrResponseWithPid();
+        writeCsv(mappingBody("aid:40126,bdbd99af-36a5-4bab-9785-e3a802d3737e,"));
+
+        var options = new DestinationMappingOptions();
+        options.setFieldName("dmrecord");
+        options.setForce(true);
+
+        service.addArchivalCollectionMappings(options);
+
```

Comment:
It'd be good to assert that the file now contains the new mappings, to demonstrate that it overwrote the existing file.